test/common/init/init_fuzz.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package test.common.init;
+
+// Structured input for init_fuzz_test.
+
+enum InitEntityAction {
+
+  // manager
+  INIT_MANAGER_CREATION = 0;
+  INIT_MANAGER_INITIALIZE = 1;
+  INIT_MANAGER_DESTRUCTION = 2;
+
+  // target
+  INIT_TARGET_READY = 3;
+  INIT_TARGET_DESTRUCTION = 4;
+
+  // watcher
+  INIT_WATCHER_DESTRUCTION = 5;
+
+  // add
+  INIT_MANAGER_ADD_TARGET = 6;
+}
+
+message InitAction {
+  InitEntityAction action = 1;
+  int32 param = 2;
+}
+
+message InitTestCase {
+  repeated InitAction actions = 1;
```

Comment:
@asraa do you think we should advocate for FuzzDataProvider here (for performance, and because the trace format is simple), or are there good reasons to keep this a proto trace format?

Reply:
`FuzzDataProvider`! TIL. Will explore and introduce

Reply:
Oh no!! I had suggested this change yesterday but I didn't know GH review process has an extra submit button.

Yeah, I personally think that if the actions are simple enough to describe, using a `FuzzedDataProvider` to chose oneof some number of actions is better. 

But if there's a strong case for the order of the scenarios bringing interesting behavior (the ordering of ready, add target, destruction) without necessarily strictly increasing code coverage, I think the proto trace format is better. I only think though that if you decide to do the proto trace format way, that you enter a good number of corpus entries.